cmd/server/transaction_storage_sqlite.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2019 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/kit/log"
+)
+
+type sqliteTransactionRepository struct {
+	db     *sql.DB
+	logger log.Logger
+}
+
+func setupSqliteTransactionStorage(logger log.Logger, path string) (*sqliteTransactionRepository, error) {
+	db, err := createSqliteConnection(logger, path)
+	if err != nil {
+		return nil, err
+	}
+	return &sqliteTransactionRepository{db, logger}, nil
+}
+
+func (r *sqliteTransactionRepository) Ping() error {
+	return r.db.Ping()
+}
+
+func (r *sqliteTransactionRepository) Close() error {
+	return r.db.Close()
+}
+
+func (r *sqliteTransactionRepository) createTransaction(t transaction) error {
+	if err := t.validate(); err != nil {
+		return fmt.Errorf("transaction=%q is invalid: %v", t.ID, err)
+	}
+
+	tx, err := r.db.Begin()
+	if err != nil {
+		return fmt.Errorf("createTransaction: tx.Begin: %v", err)
+	}
+
+	// insert transaction
+	query := `insert into transactions(transaction_id, timestamp, created_at) values (?, ?, ?);`
+	stmt, err := tx.Prepare(query)
+	if err != nil {
+		return fmt.Errorf("createTransaction: prepare: error=%v rollback=%v", err, tx.Rollback())
+	}
+	if _, err := stmt.Exec(t.ID, t.Timestamp, time.Now()); err != nil {
+		stmt.Close()
+		return fmt.Errorf("createTransaction: insert: error=%v rollback=%v", err, tx.Rollback())
+	}
+	stmt.Close()
+
+	// insert each transactionLine
+	for i := range t.Lines {
+		query = `insert into transaction_lines(transaction_id, account_id, purpose, amount, created_at) values (?, ?, ?, ?, ?);`
+		stmt, err = tx.Prepare(query)
+		if err != nil {
+			stmt.Close()
+			return fmt.Errorf("createTransaction: transaction=%q account=%q prepare: error=%v rollback=%v", t.ID, t.Lines[i].AccountId, err, tx.Rollback())
+		}
+		if _, err := stmt.Exec(t.ID, t.Lines[i].AccountId, t.Lines[i].Purpose, t.Lines[i].Amount, time.Now()); err != nil {
+			stmt.Close()
+			return fmt.Errorf("createTransaction: transaction=%q account=%q insert: error=%v rollback=%v", t.ID, t.Lines[i].AccountId, err, tx.Rollback())
+		}
+		stmt.Close()
+
+		// // Check account balance, and if we're negative by less than t.Lines[i].Amount then we need to rollback as that account
```

Comment:
Right now this creates a bit of a "chicken and egg problem" wherein some account first needs a balance in order to transfer funds to others, or we add logic to ignore the insufficient funds checks on accounts whose routing numbers aren't managed by GL. (Those accounts likely shouldn't show up in GL's account endpoints either?)